File: chattools/src/main/java/aytackydln/chattools/telegram/dto/models/InlineKeyboard.java
package aytackydln.chattools.telegram.dto.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import java.util.ArrayList;
import java.util.List;

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class InlineKeyboard implements Keyboard {
    private final List<List<KeyboardItem>> content = new ArrayList<>();

    @JsonProperty("inline_keyboard")
    public List<List<KeyboardItem>> getContent() {
        return content;
    }

    @JsonProperty
    public boolean isOneTimeKeyboard() {
        return true;
    }

    public List<KeyboardItem> addRow(){
        List<KeyboardItem> newList=new ArrayList<>();
        content.add(newList);
        return newList;
    }
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/models/InlineKeyboard.kt
package aytackydln.chattools.telegram.dto.models

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
class InlineKeyboard : Keyboard {
    @get:JsonProperty("inline_keyboard")
    val content: MutableList<MutableList<KeyboardItem>> = ArrayList()

    @get:JsonProperty
    val isOneTimeKeyboard: Boolean
        get() = true

    fun addRow(): MutableList<KeyboardItem> {
        val newList: MutableList<KeyboardItem> = ArrayList()
        content.add(newList)
        return newList
    }
}
